src/components/ui/Modal.tsx

```diff
@@ -0,0 +1,56 @@
+import { ComponentProps, useRef } from 'react';
+
+// Modal 컴포넌트 사용 예시
+// <Modal width="w-80" actionText="답변 작성하러 가기" handleAction={이벤트핸들러} cancelText="되돌아가기">아직 답변을 등록하지 않았어요.</Modal>
+
+type ModalProps = ComponentProps<'div'> & {
+  actionText: string;
+  cancelText: string;
+  width: string;
+  handleAction: () => void;
+};
```

Comment:
현재 기획중이신 디자인에서 어떠한 종류의 모달이 사용될지 (혹은 사용되지 않을지) 모르겠지만, 제 경험을 바탕으로 모달의 인터페이스를 정의한다면 다음과 같이 바꿀 것 같아요: 
```suggestion
type ModalProps = ComponentProps<'div'> & {
  actionText: string;
  cancelText?: string;
  width: string;
  onActionButtonClick: () => void;
  onCancelButtonClick?: () => void;
};
```

그리고 cancleButton의 text와 onClick핸들러가 존재한다면 버튼 두 개를, 존재하지 않는다면 액션 버튼만을 렌더링하는 형태로 구현할 것 같아요. 일반적으로는 버튼 하나짜리 컨펌 모달과 두 개 짜리 모달이 많이 쓰이더라고요! 

더불어, 현재는 cancel 버튼을 클릭할 때 단순히 모달을 닫는 동작만을 생각하셔서 해당 버튼 핸들러가 따로 없는 것 같은데… 캔슬 버튼이 클릭될 때 액션을 주입할 수 있으면 (예를 들면 입력값을 초기화하거나 하는 동작들이 있거나, 로깅이 필요할 수도 있고요) 추후 편리하지 않을까 싶어요.

Reply:
캔슬 버튼에 onClick 핸들러가 없다면(액션이 없다면) 캔슬 버튼을 렌더링하지 않는 형태로 구현할 것 같다는 말씀이신가요? 저도 구현하면서 여기에 캔슬 버튼이 필요할지 의문이 들었는데, 피드백을 받고 의문점이 더 확실해졌어요! 팀원과 의논해봐야겠습니다👍

추가적으로 캔슬 버튼에 따로 필요한 액션이 없을지도 생각해보겠습니다!